src/main/java/com/example/kanbanserv/web/ProjectController.java

```diff
@@ -45,8 +47,18 @@ public ResponseEntity<?> createNewProject(@Valid @RequestBody Project project, E
                                                   .errorDetailsList(errorList)
                                                   .build(), HttpStatus.BAD_REQUEST);
         }
-        Project project1 = projectService.saveOrUpdateProject(project);
-        return new ResponseEntity<Project>(project, HttpStatus.CREATED);
+        try{
+            Project project1 = projectService.saveOrUpdateProject(project);
+            return new ResponseEntity<Project>(project, HttpStatus.CREATED);
+        }
+        catch(Exception e) {
```

Comment:
generally, it is not recommended to catch the root class `Exception`. 

Let's say, I used a unique project identifier but `projectService.saveOrUpdateProject(...)` errors out due to some issue in `projectRepository.save(...)`, it will be caught here and we will still show `Duplicate Project Identifier` error to the user.

Solution:
Create a new Exception class with name something like `DuplicateIdentifierException` and throw that from the `saveOrUpdateProject` method and catch the same here.

Reply:
Well point made! Have added the custom exception `DuplicateIdentifierException` Please have a look.